### PR TITLE
Fix OTEL Services integration

### PR DESCRIPTION
**Pull request by Swiddis**
### Description
Fixes the unit tests and adds better debug info to the relevant test to make debugging faster in the future. Todo: the tests are still really noisy if there's an invalid integration, since all serialization tests fail if any of the serialized integrations is invalid, it should be tidied up for the rest of them.

### Issues Resolved
N/A

### Check List
- [X] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ps48 — APPROVED** [review]

**Comment by ps48** [comment]
We can wait for the tests to complete before the merge

**Comment by Swiddis** [comment]
No backport / will be applied in #1769

**Review by YANG-DB — APPROVED** [review]

**Review by paulstn — APPROVED** [review]

**Pull request merged**
